ndr_fasttransfer.c

```diff
@@ -0,0 +1,467 @@
+/*
+  OpenChange implementation.
+
+  libndr mapi support
+
+  Copyright (C) Julien Kerihuel 2015
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "libmapi/libmapi.h"
+#include "libmapi/libmapi_private.h"
+#include <ndr.h>
+#include "gen_ndr/ndr_exchange.h"
+#include "gen_ndr/ndr_property.h"
+
+#define COLOR_BLACK    "\x1b[30m"
+#define COLOR_RED      "\x1b[31m"
+#define COLOR_GREEN    "\x1b[32m"
+#define COLOR_YELLOW   "\x1b[33m"
+#define COLOR_BLUE     "\x1b[34m"
+#define COLOR_MAGENTA  "\x1b[35m"
+#define COLOR_CYAN     "\x1b[36m"
+#define COLOR_WHITE    "\x1b[37m"
+#define COLOR_BOLD     "\x1b[1m"
+#define COLOR_INVERSE  "\x1b[7m"
+#define COLOR_BOLD_OFF "\x1b[22m"
+#define COLOR_END      "\x1b[0m"
+
```

Comment:
I know that colours are fancy but they provoke reading problems using standard tools like `less`. I know I can use `less -R` and such but most of the UNIX service logs are not using colours. Sysadmins use `ccze` (https://github.com/madhouse/ccze) to colourise the log lines. We can implement a module for it.

Could we have a way to avoid using colours by default? I know they can be useful for some people, so a way to know if the display is able to print it or a configuration variable or something could be really great and an option to advance users.